core/src/main/java/org/mule/config/ChainedThreadingProfile.java

```diff
@@ -184,7 +184,7 @@ public void setWorkManagerFactory(WorkManagerFactory workManagerFactory)
     public WorkManager createWorkManager(String name, int shutdownTimeout)
     {
         // we deliberately don't instantiate the chained profile as we just want a cloned copy, not recursion
-        return workManagerFactory.createWorkManager(new ImmutableThreadingProfile(this), name, shutdownTimeout);
+        return workManagerFactory.createWorkManager(this, name, shutdownTimeout);
```

Comment:
I don't know this class in detail, but javadoc says that we should only use ImmutableThreadingProfile instances